src/Compilers/Core/Portable/PEWriter/MetadataVisitor.cs

```diff
@@ -273,6 +273,7 @@ public virtual void Visit(IMethodDefinition method)
         {
             this.Visit(method.GetReturnValueAttributes(Context));
             this.Visit(method.ReturnValueCustomModifiers);
+            this.Visit(method.RefCustomModifiers);
```

Comment:
>this.Visit(method.RefCustomModifiers); [](start = 12, length = 38)

It feels like RefCustomModifiers should be visited before ReturnValueCustomModifiers. Also, it looks like there are two more places in ReferenceIndexerBase.cs that should be adjusted in a similar way.